build-tools/installers/create-installers.targets

```diff
@@ -119,6 +119,10 @@
     <_MSBuildFiles Include="$(MSBuildSrcDir)\Xamarin.Android.Tools.JavaDocImporter.dll" />
     <_MSBuildFiles Include="$(MSBuildSrcDir)\Xamarin.Android.VisualBasic.targets" />
     <_MSBuildFiles Include="$(MSBuildSrcDir)\Xamarin.Build.AsyncTask.dll" />
+    <_MSBuildFiles Include="$(MSBuildSrcDir)\Android\x86\xdelta3" />
```

Comment:
Would it make more sense to install these files into `$(MSBuildSrcDir)\lib\x86`/etc., as we do with the native libraries?  I imagine this would require less special-casing in the future, when we need to start inserting `xdelta3` into the `.apk`...

Reply:
Well these are not lib's they are exe's. They will get fast deployed to the `/data/local/tmp` directory and then executed from that location. As such they will never get into an .apk.
I am trying to figure out if we could put it in an apk and execute these files from outside of the apk. But I suspect Android will not let us do that, so  `/data/local/tmp` is what we might need to live with.

Reply:
> I am trying to figure out if we could put it in an apk and execute these files from outside of the apk.

You *can*; that's how using `gdbserver` used to work; see the removal in commit d80b485bcbc6dff29dbd79aa2423f283dd62c60a.

The key, though, is that you need to "Jujitsu" the Android installer, and "rename" `xdelta3` to `libxdelta3.so`.  This way Android will extract & install the file into the app's lib directory.

Though there is a longevity question here; when `/manifest/application[@android:extractNativeLibs='false']` within `AndroidManifest.xml`, then `libxdelta3.so` will no longer be extracted onto the filesystem, meaning it cannot be executed.

Thus, for long-term use, we *should* rely on your suggestion of copying to `/data/local/tmp`...